File: scripts/import-jobs.js
const fs = require('fs');
const path = require('path');
const csv = require('csv-parser');

// CSV 文件路径
const csvFilePath = '/Users/charlie/Downloads/jobs.csv';
const jobsFilePath = path.join(__dirname, '../data/jobs.json');

// 存储解析后的职位数据
const jobs = [];

// 读取现有的 jobs.json
let existingJobs = [];
try {
  const content = fs.readFileSync(jobsFilePath, 'utf-8');
  existingJobs = JSON.parse(content);
  console.log(`已加载 ${existingJobs.length} 个现有职位`);
} catch (error) {
  console.log('未找到现有职位数据，将创建新文件');
}

// 解析 CSV 并转换为 Job 格式
fs.createReadStream(csvFilePath)
  .pipe(csv())
  .on('data', (row) => {
    // 跳过空行
    if (!row.title || row.title.trim() === '') {
      return;
    }

    // 提取公司名称（从 id 列或单独的公司名称列）
    let companyName = row.id || '未知公司';

    // 如果 id 是数字，说明这是旧格式，公司名称可能在其他地方
    if (!isNaN(companyName)) {
      // 尝试从描述中提取公司名称
      const descLines = (row.description || '').split('\n');
      for (const line of descLines) {
        if (line.includes('公司') || line.includes('介绍')) {
          companyName = line.replace(/【|】|公司介绍|公司/g, '').trim();
          break;
        }
      }
      if (companyName === row.id) {
        companyName = '未知公司';
      }
    }

    // 解析技能要求
    let requirements = '';
    if (row.required_skills) {
      try {
        const skills = row.required_skills
          .replace(/[{}]/g, '')
          .split(',')
          .filter(s => s.trim())
          .join('、');
        if (skills) {
          requirements = `技能要求：${skills}\n\n`;
        }
      } catch (e) {
        // 忽略解析错误
      }
    }

    // 从描述中提取地点、工作方式、职位类型
    const description = row.description || '';
    let baseLocation = '未指定';
    let workType = 'hybrid'; // 默认混合
    let employmentType = 'both'; // 默认都可以

    // 提取地点
    const locationMatch = description.match(/工作地点[：:]\s*([^\n]+)/i) ||
                         description.match(/地点[：:]\s*([^\n]+)/i) ||
                         description.match(/(北京|上海|深圳|杭州|广州|成都|苏州|南京|武汉|西安|济南)/);
    if (locationMatch) {
      baseLocation = locationMatch[1] || locationMatch[0];
      baseLocation = baseLocation.trim().split(/[，,、]/)[0]; // 取第一个地点
    }

    // 提取工作方式
    if (description.match(/线下|现场|到岗/i)) {
      workType = 'offline';
    } else if (description.match(/线上|远程/i)) {
      workType = 'online';
    }

    // 提取职位类型
    if (description.match(/实习/i) && !description.match(/全职/i)) {
      employmentType = 'intern';
    } else if (description.match(/全职/i) && !description.match(/实习/i)) {
      employmentType = 'full-time';
    }

    // 提取联系方式
    let contact = '';
    const emailMatch = description.match(/([a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,})/);
    const wechatMatch = description.match(/微信[：:]\s*([^\s\n]+)/i) ||
                       description.match(/wechat[：:]\s*([^\s\n]+)/i);

    if (emailMatch) {
      contact = emailMatch[1];
    } else if (wechatMatch) {
      contact = `微信: ${wechatMatch[1]}`;
    }

    // 创建 Job 对象
    const job = {
      id: Date.now().toString() + Math.random().toString(36).substr(2, 9),
      companyName: companyName,
      title: row.title.trim(),
      description: description.trim(),
      requirements: requirements + (description.trim()),
      contact: contact || '请查看职位详情',
      baseLocation: baseLocation,
      workType: workType,
      employmentType: employmentType,
      createdAt: new Date().toISOString()
    };

    jobs.push(job);
  })
  .on('end', () => {
    console.log(`\n成功解析 ${jobs.length} 个职位`);

    // 合并现有职位和新职位
    const allJobs = [...existingJobs, ...jobs];

    // 写入文件
    fs.writeFileSync(jobsFilePath, JSON.stringify(allJobs, null, 2));
    console.log(`已将 ${allJobs.length} 个职位写入 ${jobsFilePath}`);

    // 显示前 3 个职位作为示例
    console.log('\n导入的职位示例：');
    jobs.slice(0, 3).forEach((job, index) => {
      console.log(`\n${index + 1}. ${job.companyName} - ${job.title}`);
      console.log(`   地点: ${job.baseLocation}`);
      console.log(`   类型: ${job.employmentType === 'intern' ? '实习' : job.employmentType === 'full-time' ? '全职' : '实习/全职'}`);
      console.log(`   工作方式: ${job.workType === 'online' ? '线上' : job.workType === 'offline' ? '线下' : '混合'}`);
    });
  })
  .on('error', (error) => {
    console.error('导入失败:', error);
  });

File: scripts/import-jobs.py
#!/usr/bin/env python3
import csv
import json
import re
from datetime import datetime
import random
import string

# 文件路径
csv_file = '/Users/charlie/Downloads/jobs.csv'
jobs_file = '/Users/charlie/pluslink/data/jobs.json'

# 读取现有职位
try:
    with open(jobs_file, 'r', encoding='utf-8') as f:
        existing_jobs = json.load(f)
    print(f"已加载 {len(existing_jobs)} 个现有职位")
except:
    existing_jobs = []
    print("未找到现有职位数据，将创建新文件")

jobs = []

# 读取 CSV
with open(csv_file, 'r', encoding='utf-8-sig') as f:
    reader = csv.DictReader(f)

    for row in reader:
        # 获取 id 列作为公司名称
        company_name = row.get('id', '').strip()

        # 跳过空行
        if not company_name or not row.get('title', '').strip():
            continue

        # 如果 id 是纯数字，跳过这一行（删除数字 id 的职位）
        if company_name.isdigit():
            print(f"跳过数字 ID: {company_name} - {row.get('title', '')}")
            continue

        # 提取其他字段
        title = row.get('title', '').strip()
        description = row.get('description', '').strip()
        required_skills = row.get('required_skills', '').strip()

        # 解析技能要求
        requirements = ''
        if required_skills:
            try:
                skills = required_skills.replace('{', '').replace('}', '').split(',')
                skills = [s.strip() for s in skills if s.strip()]
                if skills:
                    requirements = f"技能要求：{'、'.join(skills)}\n\n"
            except:
                pass

        # 从描述中提取地点
        base_location = '未指定'
        location_patterns = [
            r'工作地点[：:]\s*([^\n]+)',
            r'地点[：:]\s*([^\n]+)',
            r'(北京|上海|深圳|杭州|广州|成都|苏州|南京|武汉|西安|济南|重庆|天津|长沙|郑州|厦门|青岛|大连|宁波|合肥|福州)'
        ]
        for pattern in location_patterns:
            match = re.search(pattern, description, re.IGNORECASE)
            if match:
                base_location = match.group(1) if match.lastindex else match.group(0)
                base_location = base_location.strip().split('，')[0].split(',')[0]
                break

        # 提取工作方式
        work_type = 'hybrid'
        if re.search(r'线下|现场|到岗', description, re.IGNORECASE):
            work_type = 'offline'
        elif re.search(r'线上|远程', description, re.IGNORECASE):
            work_type = 'online'

        # 提取职位类型
        employment_type = 'both'
        has_intern = bool(re.search(r'实习', description, re.IGNORECASE))
        has_fulltime = bool(re.search(r'全职', description, re.IGNORECASE))

        if has_intern and not has_fulltime:
            employment_type = 'intern'
        elif has_fulltime and not has_intern:
            employment_type = 'full-time'

        # 提取联系方式
        contact = ''
        email_match = re.search(r'([a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,})', description)
        wechat_match = re.search(r'(?:微信|wechat)[：:]\s*([^\s\n]+)', description, re.IGNORECASE)

        if email_match:
            contact = email_match.group(1)
        elif wechat_match:
            contact = f"微信: {wechat_match.group(1)}"
        else:
            contact = '请查看职位详情'

        # 生成唯一 ID
        job_id = str(int(datetime.now().timestamp() * 1000)) + ''.join(random.choices(string.ascii_lowercase + string.digits, k=9))

        # 创建职位对象
        job = {
            'id': job_id,
            'companyName': company_name,
            'title': title,
            'description': description,
            'requirements': requirements + description,
            'contact': contact,
            'baseLocation': base_location,
            'workType': work_type,
            'employmentType': employment_type,
            'createdAt': datetime.now().isoformat()
        }

        jobs.append(job)
        print(f"✓ {company_name} - {title}")

print(f"\n成功解析 {len(jobs)} 个职位")

# 合并并写入
all_jobs = existing_jobs + jobs
with open(jobs_file, 'w', encoding='utf-8') as f:
    json.dump(all_jobs, f, ensure_ascii=False, indent=2)

print(f"已将 {len(all_jobs)} 个职位写入 {jobs_file}")

# 显示统计
print("\n公司统计：")
companies = {}
for job in jobs:
    company = job['companyName']
    companies[company] = companies.get(company, 0) + 1

for company, count in sorted(companies.items(), key=lambda x: x[1], reverse=True):
    print(f"  {company}: {count} 个职位")
